components/bookshelf/discover/SearchBox.tsx

```diff
@@ -33,17 +30,23 @@ const SearchBox = ({ setIsSearching }: SearchBoxProps) => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [])
 
-  const handleChange = (e) => {
+  const handleChange = (e: {
+    target: { value: React.SetStateAction<string> }
+  }) => {
     setSearchInput(e.target.value)
   }
 
-  const handleSubmit = (e) => {
+  const handleSubmit = (e: { preventDefault: () => void }) => {
```

Comment:
```suggestion
  const handleSubmit = (e: React.FormEvent<HTMLInputElement>) => {
```

Reply:
Where did I get that type? O hovered over the form `onSubmit` function - 
![image](https://user-images.githubusercontent.com/780659/191928244-89a019e2-9116-44ef-9baa-ce68f617c654.png)

Reply:
this one works 
![image](https://user-images.githubusercontent.com/91857581/191931183-ea481c90-59db-4eee-a1a0-2ded7b4130d2.png)